src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/TypeNameParser.NativeAot.cs

```diff
@@ -214,10 +214,9 @@ private bool CheckTopLevelAssemblyQualifiedName()
                 if (type is null && _ignoreCase && !_extensibleParser)
                 {
                     // Return the first name that matches. Which one gets returned on a multiple match is an implementation detail.
-                    string lowerName = nestedTypeNames[i].ToLowerInvariant();
                     foreach (Type nt in declaringType.GetNestedTypes(BindingFlags.NonPublic | BindingFlags.Public))
                     {
-                        if (nt.Name.ToLowerInvariant() == lowerName)
```

Comment:
This is changing semantics of the comparison. Do we really have an analyzer that suggests people blindly do this?

```csharp
using System;

// These are both lower case Greek Sigmas
string a = "σ";
string b = "ς";

// Prints True
Console.WriteLine(a.Equals(b, StringComparison.InvariantCultureIgnoreCase));

string lowerA = a.ToLowerInvariant();
string lowerB = b.ToLowerInvariant();

// Prints False
Console.WriteLine(lowerA == lowerB);
```

Reply:
Cc @carlossanlop @buyaa-n - this analyzer was added in #6720

Reply:
@carlossanlop @buyaa-n @MichalStrehovsky 

> this analyzer was added in https://github.com/dotnet/roslyn-analyzers/pull/6720

Yes, and it's been implemented incorrectly.... as has https://github.com/dotnet/runtime/issues/78606

See my comment explaining why:-
https://github.com/dotnet/roslyn-analyzers/pull/6720#issuecomment-1653051447

Reply:
Yeah, that's a good point.  I had started fixing the warnings between other tasks but stopped when I realized that the Contains analyzer is also complaining about assemblies that are multitargeted and therefore look like they will need lots of suppressions and was going to ask if there was a better to disable them for those assemblies than going line by line.

Reply:
> This is changing semantics of the comparison.

Created an [issue](https://github.com/dotnet/runtime/issues/89740) for fix.

> See my comment explaining why:-
> https://github.com/dotnet/roslyn-analyzers/pull/6720#issuecomment-1653051447

Thanks @Bellarmine-Head your comments might disregarded because the PR/issue was closed or unrelated, feel free to add your comments/suggestions into the [dedicated issue](https://github.com/dotnet/runtime/issues/89740) 


> I had started fixing the warnings between other tasks but stopped when I realized that the Contains analyzer is also complaining about assemblies that are multitargeted and therefore look like they will need lots of suppressions and was going to ask if there was a better to disable them for those assemblies than going line by line.

We can lower severity of CA1862: `Prefer using 'StringComparer'/'StringComparison' to perform case-insensitive string comparisons` until the bug is fixed. For other analyzer fixes/failures looks valid to me, I will try to fix them and disable at project level if needed